aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+The following action method uses the ``Ok`` and ``NotFound`` helper methods:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 28-38
+  :emphasize-lines: 8,10
+  :dedent: 8
+
+A JSON-formatted response will be returned unless another format was requested and the server can return the requested format. You can use a tool like `Fiddler <http://www.telerik.com/fiddler>`_ to create a request that includes an Accept header, and specify another format. In that case, if the server has a *formatter* that matches the requested format, the results will be returned in that format.
+
+.. image:: formatting/_static/fiddler-composer.png
+
+In the above screenshot, the Fiddler Composer has been used to generate a request, specifying ``Accept: application/xml``. By default, ASP.NET Core MVC only supports JSON, so even when another format is specified, the result returned is still JSON-formatted. You'll see how to add additional formatters in the next section.
+
+Your controller actions can return model objects, in which case ASP.NET MVC will automatically create an ``ObjectResult`` for you that wraps the object. The client will get the appropriately formatted serialized object. If the object being returned is ``null``, then the framework will return a ``204 No Content`` response.
+
+Returning an object type:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 40-45
+  :emphasize-lines: 3
+  :dedent: 8
+
+In the sample, a request for a valid author alias will receive a 200 OK response with the author's data. A request for an invalid alias will receive a 204 No Content response. Screenshots showing the response in XML and JSON formats are shown below.
+
+Content Negotiation Process
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Content *negotiation* only takes place if an ``Accept`` header appears in the request. If the request contains an Accept header the framework will enumerate through the list of formatters until it finds one that can serialize the object being returned into the requested format. If no formatter is found that can support the type to be returned, then the framework will send a 406 Not Acceptable response code. If the request specifies XML, but the XML formatter has not been configured, then the JSON formatter will be used. If no header is given, the first formatter that can handle the object to be returned will be used to serialize the response. In this case, there isn't any negotiation taking place - the server is determining what format it will use.
```

Comment:
>Content *negotiation* only takes place if an ``Accept`` header appears in the request. If the request contains an Accept header the framework will enumerate through the list of formatters until it finds one that can serialize the object being returned into the requested format. If no formatter is found that can support the type to be returned, then the framework will send a 406 Not Acceptable response code. If the request specifies XML, but the XML formatter has not been configured, then the JSON formatter will be used. If no header is given, the first formatter that can handle the object to be returned will be used to serialize the response. In this case, there isn't any negotiation taking place - the server is determining what format it will use.

When a request contains an accept header, the framework will enumerate the media types in the accept header in preference order and will try to find a formatter that can produce a response in one of the formats specified by the accept header. In case no formatter is found that can satisfy the clients request, the framework will try to find the first formatter that can produce a response unless the developer has configured the option on MvcOptions to return Not acceptable instead. 

If the Accept header contains */*, it will be ignored unless RespectBrowserAcceptHeader is set to true on the MVcOptions